server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,745 @@
+                            f.write("\t* Prefix directory, .prefix, is not a directory!\n")
+            f.write("\nend-{}: archive hierarchy: {}\n".format(self.config.TS[4:], self.path))
+            f.close()
+        return cnt
+
+    def output_format(self, f, controller, controller_val, cnt):
+        f.write("\t  ++++++++++\n")
+        for val in sorted(self.things_to_check[controller_val]._dict[controller]):
+            f.write("\t  {}\n".format(val))
+        f.write("\t  ----------\n")
+        cnt += 1
+        return cnt
+
+
+def verify_subdirs(hier, controller, directories):
+    linkdirs = sorted(hier.config.LINKDIRS.split(' '))
+    if directories:
+        for dirent in directories:
+            if all([os.path.exists(os.path.join(hier.path, controller, dirent)), dirent != '_QUARANTINED', not dirent.startswith('WONT-INDEX')]):
+                if dirent not in linkdirs:
+                    hier.add_unexpected_dirs(controller, dirent)
+    else:
+        hier.set_status(controller, "subdirs")
+
+    return 0
+
+def verify_prefixes(hier, controller):
+    prefix_dir = os.path.join(hier.path, controller, ".prefix")
+    if not os.path.exists(prefix_dir):
+        return
+    if not os.path.isdir(prefix_dir):
+        hier.set_status(controller, "prefix_dir")
+        return
+
+    prefixes = glob.iglob(os.path.join(prefix_dir,"*"))
+
+    for prefix in prefixes:
+        base_prefix = os.path.basename(prefix)
+        if not base_prefix.startswith('prefix.') and not base_prefix.endswith('.prefix'):
+            hier.add_non_prefixes(controller, base_prefix)
+        elif base_prefix.startswith('prefix.'):
+            hier.add_wrong_prefixes(controller, base_prefix)
+
+    return
+
+def verify_tarball_names(hier, controller):
+    pass
+
+def verify_archive(hier):
+    controllers = glob.iglob(os.path.join(hier.path, '*'))
+
+    for controller in controllers:
+        if os.path.isdir(controller):
+            hier.add_controller(os.path.basename(controller))
+        else:
+            hier.add_bad_controller(os.path.basename(controller))
+
+    # now check each "good" controller and get the tarballs it contains
+    for controller in hier.controllers:
+        direct_entries = glob.iglob(os.path.join(hier.path, controller, '*'))
+        hidden_entries = glob.glob(os.path.join(hier.path, controller, '.*'))
+        controller_subdir=list()
+        for item in direct_entries:
+            base_item = os.path.basename(item)
+            if os.path.isdir(item) and base_item != ".prefix":
+                controller_subdir.append(base_item)
+            elif os.path.islink(item):
+                symlink_item = f"{base_item} -> {os.path.realpath(item)}"
+                hier.add_unexpected_symlinks(controller, symlink_item)
+            elif all([os.path.isfile(item), not base_item.endswith('.tar.xz'), not base_item.endswith('.tar.xz.md5')]):
+                if hidden_entries:
```

Comment:
The hidden entries stuff should be taken out of this loop and moved after line 193: you need to check for `.prefix` which is OK, but anything else in the `hidden_entries` list is an unexpected object.

Reply:
Ok sure